pkg/proc/evalop/evalcompile.go

```diff
@@ -169,6 +169,31 @@ func (ctx *compileCtx) compileAST(t ast.Expr) error {
 				return ctx.compileUnary(node.X, &Select{node.Sel.Name})
 			}
 
+		case *ast.CallExpr:
+			ident, ok := x.Fun.(*ast.SelectorExpr)
+			if ok {
+				f, ok := ident.X.(*ast.Ident)
+				if ok && f.Name == "runtime" && ident.Sel.Name == "frame" {
+					switch arg := x.Args[0].(type) {
+					case *ast.BasicLit:
+						fr, err := strconv.ParseInt(arg.Value, 10, 8)
+						if err != nil {
+							return err
+						}
+						// Update scope to requested frame.
+						ctx.pushOp(&UpdateScopeFrame{Frame: fr})
```

Comment:
I think it's better to add a `Frame` parameter to `PushLocal` and then do `if op.Frame != 0` in `case evalop.PushLocal`. To do a proper UpdateScopeFrame op it would also have to update spoff/bpoff/fboff (with the way it's used here it doesn't matter, because it's rolled back immediately, but I think we'll forget that it only works in specific circumstances).

Reply:
Good point, and a lot cleaner. Done.